src/foam/u2/crunch/CrunchController.js

```diff
@@ -89,14 +90,24 @@ foam.CLASS({
       }
       return this.getCapabilities(capabilityId).then(capabilities => {
         // Map capabilities to CapabilityWizardSection objects
-        return Promise.all(capabilities.filter(
-          cap => !! cap.of
-        ).map(
-          cap => this.CapabilityWizardlet.create({
-            capability: cap
-          }).updateUCJ()
-        ));
-      }).then(sections => {
+        return Promise.all([
+
+          // Continue passing capabilities to next callback
+          Promise.resolve(capabilities),
+
+          // Create capability sections
+          Promise.all(capabilities.filter(
+            cap => !! cap.of
+          ).map(
+            cap => this.CapabilityWizardlet.create({
+              capability: cap
+            }).updateUCJ()
+          ))
+
+        ]);
+      }).then(capabilitiesSectionsTuple => {
+        let capabilities = capabilitiesSectionsTuple[0];
+        let sections = capabilitiesSectionsTuple[1];
```

Comment:
possible out of bounds?

Reply:
shouldn't be, call above to Promise.all always returns an array of 2 elements. I'll add a comment to document this